### ISSUE TITLE
CupertinoSwitch haptic feedback triggered way too often

### ISSUE THREAD

**Issue by knopp** [flutter/flutter#27497]
Introduced [here](https://github.com/flutter/flutter/commit/9f20bd6cb91bb61162ff60ba2ac84944f74b6e9b#diff-7fd54900390d62f0b4354d669adebcf4). The commit invokes haptic feedback when the onChanged *callback* is updated (which is every time widget hierarchy is rebuilt) instead of invoking haptic feedback when the actual value is changed.

@jslavitz 

```
[✓] Flutter (Channel master, v1.2.1-pre.67, on Mac OS X 10.14 18A389, locale en-US)
    • Flutter version 1.2.1-pre.67 at /Users/Matej/Projects/flutter/forked/flutter
    • Framework revision 42bbb441dd (3 months ago), 2018-11-21 16:40:41 +0100
    • Engine revision 15f2b92cce
    • Dart version 2.1.1 (build 2.1.1-dev.3.2 71bee8f05e)

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).
      If Android SDK has been installed to a custom location, set ANDROID_HOME to that location.
      You may also want to add it to your PATH environment variable.


[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[!] IntelliJ IDEA Community Edition (version 2018.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[✓] VS Code (version 1.30.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.22.3

[✓] Connected device (1 available)
    • Matej’s iPhone X • f7e7ead1f9b5f3e2574cfbd9bc264fbcd5772a6c • ios • iOS 12.1.2
```
